### PR TITLE
add webview ui

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: intuita-webview/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (grzpab) [inline intuita-webview/.gitignore:1]: can it use the global `.gitignore`?

File: src/utilities.ts [hunk 2]
```diff
@@ -41,3 +41,13 @@ export const debounce = <R>(callback: (...args: any[]) => R, ms: number) => {
 };
 
 export const singleQuotify = (str: string) => `'${str}'`;
+
+export function getNonce() {
```
Comment (grzpab) [inline src/utilities.ts:45]: isn't better to use `Crypto.getRandomValues()`?

File: src/extension.ts [hunk 3]
```diff
@@ -134,11 +140,71 @@ export async function activate(context: vscode.ExtensionContext) {
...
+	 * User
+	 */
 	const globalStateAccountStorage = new GlobalStateAccountStorage(
 		context.globalState,
 	);
 	const userService = new UserService(globalStateAccountStorage);
 
+	const intuitaWebviewProvider = new IntuitaPanel(context);
+	const view = vscode.window.registerWebviewViewProvider(
+		'intuita-webview',
```
Comment (grzpab) [inline src/extension.ts:155]: can you check if it works with `vsce package` command?

File: src/extension.ts [hunk 4]
```diff
@@ -134,11 +140,71 @@ export async function activate(context: vscode.ExtensionContext) {
...
+				try {
+					const codec = buildTypeCodec({
+						title: t.string,
+						body: t.string,
+					});
+
+					const decoded = codec.decode(arg0);
+
+					if (decoded._tag === 'Right') {
+						sourceControl.createIssue(decoded.right);
```
Comment (grzpab) [inline src/extension.ts:179]: is it an async method?

File: src/extension.ts [hunk 5]
```diff
@@ -134,11 +140,71 @@ export async function activate(context: vscode.ExtensionContext) {
...
+					}
+				} catch (e) {
+					// @TODO prettify error messages
```
Comment (grzpab) [inline src/extension.ts:182]: is this comment still relevant?

File: intuita-webview/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (grzpab) [inline intuita-webview/tsconfig.json:1]: I think it should extend the existing root-level `tsconfig.json`

File: intuita-webview/src/utilities/vscode.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+		if (this.vsCodeApi) {
+			return this.vsCodeApi.setState(newState);
+		} else {
+			localStorage.setItem('vscodeState', JSON.stringify(newState));
+			return newState;
+		}
+	}
+}
+
+export const vscode = new VSCodeAPIWrapper();
```
Comment (grzpab) [inline intuita-webview/src/utilities/vscode.ts:39]: I would create dependencies in the main file

File: intuita-webview/src/react-app-env.d.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+/// <reference types="react-scripts" />
```
Comment (grzpab) [inline intuita-webview/src/react-app-env.d.ts:1]: do you know why this is needed?

File: intuita-webview/.gitignore (hunk 1 above)
Comment (DmytroHryshyn) [reply intuita-webview/.gitignore:1]: i think i can just remove this nested .gitignore

File: src/extension.ts (hunk 3 above)
Comment (DmytroHryshyn) [reply src/extension.ts:155]: works fine

File: intuita-webview/src/react-app-env.d.ts (hunk 8 above)
Comment (DmytroHryshyn) [reply intuita-webview/src/react-app-env.d.ts:1]: this is needed to include react-scripts type declarations (this is autogenerated create-react-app file )

File: src/extension.ts (hunk 3 above)
Comment (DmytroHryshyn) [reply src/extension.ts:156]: @grzpab i've also made updates to `vscodeignore`, excluded node_module form webview

File: intuita-webview/src/utilities/vscode.ts (hunk 7 above)
Comment (DmytroHryshyn) [reply intuita-webview/src/utilities/vscode.ts:39]: It has to be a signleton, so i had to create instance right in vscode.ts

**Review by grzpab — APPROVED** [review]

**Pull request merged**